lib/dassets.rb

```diff
@@ -10,21 +11,34 @@ def self.configure(&block)
   end
 
   def self.init
-    @asset_files ||= {}
+    @asset_files  ||= {}
+    @source_files   = SourceFiles.new(self.config.sources)
   end
 
   def self.[](digest_path)
     @asset_files[digest_path] ||= AssetFile.new(digest_path)
   end
 
-  def self.source_list
-    SourceList.new(self.config.sources)
+  def self.source_files
+    @source_files
   end
 
-  module SourceList
+  module SourceFiles
+
     def self.new(sources)
-      sources.inject([]){ |list, source| list += source.files }
+      # use a hash to store the source files so in the case two source files
+      # have the same digest path, the last one *should* be correct since it
+      # was last to be configured
+      hash = {}
+      sources.each do |source|
+        source.files.each do |file_path|
+          s = SourceFile.new(file_path)
+          hash[s.digest_path] = s
+        end
+      end
+      hash
```

Comment:
@jcredding do you have a better suggestion for turning this list sources with lists of files into a hash of source files?  I tried this thinking it would be the most performant (ie not doing an inject with many merges or nested injects).  Thoughts?